### PR TITLE
Update data generating process.

### PR DESCRIPTION
**Pull request by zmccaw-insitro**
Add an option to vary the proportion of variants in a gene that are causal when simulating data. Updated package passes dev check.

### CODE REVIEW

File: R/generate_data.R [hunk 1]
```diff
@@ -297,8 +343,9 @@ DGP <- function(
   maf_range = c(0.005, 0.010),
   method = "none",
   n = 100,
-  p_dmv = 0.33,
-  p_ptv = 0.33,
+  p_dmv = 0.40,
+  p_ptv = 0.10,
```
Comment (ctk3b) [inline R/generate_data.R:347]: What was the rationale for updating these two defaults?

File: R/generate_data.R [hunk 2]
```diff
@@ -80,6 +80,38 @@ GenGeno <- function(
...
+#' @return List containing the (n x snps) genotype matrix "geno" and the
+#' (snps x 1) annotation vector "anno".
+FilterGenos <- function(
+  anno,
+  geno,
+  prop_causal = 1.0
+) {
+  
+  p_noncausal <- 1.0 - prop_causal
+  which_drop <- stats::rbinom(length(anno), size = 1, prob = p_noncausal)
```
Comment (ctk3b) [inline R/generate_data.R:99]: Could you document why we use `binom` here to perform the selection?

**Review by ctk3b — APPROVED** [review]
LGTM % minor questions and docs

File: R/generate_data.R (hunk 1 above)
Comment (zmccaw-insitro) [reply R/generate_data.R:353]: Thanks for the question, these are based on the frequencies in UK Biobank. Updated the documentation to note this.

File: R/generate_data.R (hunk 2 above)
Comment (zmccaw-insitro) [reply R/generate_data.R]: This was the first approach that came to mind when writing the function, but your completely right, binom is not needed here. Updated to simplify.

**Pull request merged**